scripts/azureml-assets/azureml/assets/release/model_publish_utils/mlflow_model_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import logging
+import os
+import azureml.evaluate.mlflow as mlflow
+from mlflow.models import ModelSignature
+from sqlalchemy import true
+from transformers import  AutoTokenizer, AutoConfig, pipeline
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoModelForMaskedLM,
+    AutoModelForTokenClassification,
+    AutoModelForQuestionAnswering,
+    AutoModelWithLMHead
+)
+
+class MLFlowModelUtils:
+
+    def __init__(self, name, task_name, flavor, model_dir):
+        self.name = name
+        self.task_name = task_name
+        self.model_dir = model_dir
+        self.flavor = flavor
+
+    def _convert_to_mlflow_hftransformers(self):
+        config = AutoConfig.from_pretrained(self.name)
+        misc_conf = {"task_type": self.task_name}
+        task_model_mapping = {
+            "multiclass": AutoModelForSequenceClassification,
+            "multilabel": AutoModelForSequenceClassification,
+            "fill-mask": AutoModelForMaskedLM,
+            "ner": AutoModelForTokenClassification,
+            "question-answering": AutoModelForQuestionAnswering,
+            "summarization": AutoModelWithLMHead,
```

Comment:
for `AutoModelWithLMHead`, we get below deprecation warning: 

```
/home/azureuser/miniconda3/envs/mp/lib/python3.8/site-packages/transformers/models/auto/modeling_auto.py:806: FutureWarning: The class `AutoModelWithLMHead` is deprecated and will be removed in a future version. Please use `AutoModelForCausalLM` for causal language models, `AutoModelForMaskedLM` for masked language models and `AutoModelForSeq2SeqLM` for encoder-decoder models.
```

can we just go with just with [AutoModel](https://huggingface.co/transformers/v3.0.2/model_doc/auto.html) for all task types? 

cc: @ghyadav

Reply:
The output of an AutoModel will be raw hidden states without any specific head on top. Our current scoring script uses HF pipelines, which need a specialized model(with the appropriate head); they don't work with AutoModels. I think @ghyadav's library also needs a model with a specialized head, as a few of the tasks are implemented using pipelines. We'll need to go ahead with this approach itself.
As for the warning, I think it can be avoided by using AutoModelForSeq2SeqLM/AutoModelForCausalLM instead of AutoModelWithLMHead. We'll test that change out and update the PR once it's done.